AllJoyn/Platform/BridgeHost/App.xaml.cs

```diff
@@ -0,0 +1,359 @@
+
+            if (isStartupTask ||
+                IsSupportedAppServiceClient(triggerDetails.CallerPackageFamilyName))
+            {
+                appServiceDeferral = taskInstance.GetDeferral();
+                taskInstance.Canceled += TaskInstance_Canceled;
+
+                var appServiceDetails = triggerDetails as AppServiceTriggerDetails;
+                Debug.WriteLine("New Connection from " + appServiceDetails.CallerPackageFamilyName);
+                appServiceConnection = appServiceDetails.AppServiceConnection;
+                appServiceConnection.RequestReceived += AppServiceConnection_RequestReceived;
+                appServiceConnection.ServiceClosed += AppServiceConnection_ServiceClosed;
+            }
+
+            if (isStartupTask)
+            {
+                isStartUpTaskRunning = true;
+            }
+            else
+            {
+                StartOpenT2TAppServiceAnchorProcess(); // Background process is not yet launched. Launch it so that the reference is kept alive.
+            }
+
+            // Onboard the devices that were onboarded previously.
+            await OnboardRegisteredDevices();
+        }
+
+        private bool IsSupportedAppServiceClient(string callerPackageFamilyName)
+        {
+            return callerPackageFamilyName.Equals("org.opent2t.console_h35559jr9hy9m", StringComparison.OrdinalIgnoreCase) || // OpenT2T Console app
+                callerPackageFamilyName.Equals("OpenT2TAllJoynBridgeUWPTestClient_a6049pvczh34w", StringComparison.OrdinalIgnoreCase); // Test app
+        }
+
+        private async void AppServiceConnection_RequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            AppServiceDeferral msgDeferral = args.GetDeferral();
+            string response = "failed";
+
+            var onboardingData = args.Request.Message;
+            if ((onboardingData != null) && onboardingData.Keys.Contains("command"))
+            {
+                string command = onboardingData["command"] as string;
+                Debug.WriteLine("Command Received: " + command);
+                if (command.Equals("AddDevice"))
+                {
+                    try
+                    {
+                        await AddDeviceToAllJoynAsync(onboardingData);
+                        response = "succeeded";
+                    }
+                    catch (Exception e)
+                    {
+                        response = e.Message;
+                    }
+                }
+                else if ((command != null) && command.Equals("healthcheck"))
+                {
+                    response = "healthy";
+                }
+            }
+            ValueSet valueSet = new ValueSet();
+            valueSet.Add("response", response);
+            await args.Request.SendResponseAsync(valueSet);
+            msgDeferral.Complete();
```

Comment:
#ByDesign

---
In reply to: [75204755](https://github.com/tjaffri/msiot-samples/pull/9#discussion_r75204755) [](ancestors = 75204755,75185000)